docs/http/http-metrics.md

```diff
@@ -701,4 +705,173 @@ Tracing instrumentations that do so, MUST also set `http.request.method_original
+
+| Value  | Description |
+|---|---|
+| `active` | active state. |
+| `idle` | idle state. |
+<!-- endsemconv -->
+
+### Metric: `http.client.connection.duration`
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/metrics/api.md#instrument-advisory-parameters)
+of `[ 0.01, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 30, 60, 120, 300 ]`.
+
+**Status**: [Experimental][DocumentStatus]
+
+This metric is optional.
+
+<!-- semconv metric.http.client.connection.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `http.client.connection.duration` | Histogram | `s` | The duration of the successfully established outbound HTTP connections. |
+<!-- endsemconv -->
+
+<!-- semconv metric.http.client.connection.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| [`network.peer.address`](../attributes-registry/network.md) | string | Peer address of the network connection - IP address or Unix domain socket name. | `10.1.2.80`; `/tmp/my.sock` | Recommended |
+| [`network.protocol.version`](../attributes-registry/network.md) | string | HTTP protocol version of the connection in the connection pool. [1] | `1.1`; `2`; `3` | Recommended |
+| [`server.address`](../attributes-registry/server.md) | string | Host identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) HTTP request is sent to. [2] | `example.com`; `10.1.2.80`; `/tmp/my.sock` | Required |
+| [`server.port`](../attributes-registry/server.md) | int | Port identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) HTTP request is sent to. [3] | `80`; `8080`; `443` | Conditionally Required: [4] |
+| [`url.scheme`](../attributes-registry/url.md) | string | The [URI scheme](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) component identifying the used protocol. | `http`; `https`; `ftp` | Recommended |
+
+**[1]:** HTTP 1.0 and 1.1 requests share connections in the connection pool and are both reported as version `1.1`. So, the `network.protocol.version` value reported on connection metrics is different than the one reported on request-level metrics or spans for HTTP 1.0 requests.
+
+**[2]:** If an HTTP client request is explicitly made to an IP address, e.g. `http://x.x.x.x:8080`, then `server.address` SHOULD be the IP address `x.x.x.x`. A DNS lookup SHOULD NOT be used.
+
+**[3]:** When observed from the client side, and when communicating through an intermediary, `server.port` SHOULD represent the server port behind any intermediaries, for example proxies, if it's available.
+
+**[4]:** If not the default (`80` for `http` scheme, `443` for `https`).
+<!-- endsemconv -->
+
+### Metric: `http.client.request.time_in_queue`
```

Comment:
This metric is very specific to the .NET Connection pool implementation, I find it unlikely that other stacks can adapt it.

Reply:
We have an example of a similar (and far from perfect) metric here 
https://github.com/open-telemetry/semantic-conventions/blob/f6f4ead93651fa26652d11605beffc12873d1738/docs/database/database-metrics.md?plain=1#L168

It's pretty common to report at least some metrics for connection pool: proof [reactor-netty in Java](https://projectreactor.io/docs/netty/1.1.10-SNAPSHOT/reference/index.html#_metrics_5),  [okhttp in Java](https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/okhttp3/OkHttpConnectionPoolMetrics.java)

So I think this metric can be applied to any other technologies that use connection pooling (http, db clients).

Reply:
Having said this, I think it has a future as a `http.client.connection.pool.wait_time` or even `connection.cleint.pool.wait_time`. 

If we could go back in time, do you think we'd name this one as  `dotnet.http.client.request.time_in_queue` ?

Reply:
To elaborate on my concern, here are two strategies for dispatching pooled connections to requests I have encountered so far, https://github.com/dotnet/runtime/issues/93832#issuecomment-1775459378 llustrates the differences nicely:
- Coupled: when a request comes in and there is no connection is availble, the request spans a new connection attempt (DNS, TCP, TLS) and waits for it to complete. Such a connection attempt can be represented as a subspan of the request span. If a connection is available, there is no "wait time", the connection can be immediately associated to the request. This used to be the .NET strategy until 6.0, and AFAIK most HTTP stacks do this.
- Decoupled: connection creation is an independent background activity while requests are waiting in a queue until a connection is ready. This is the .NET strategy since 6.0. I don't know if there are any other stacks doing this.

Uniting the semantics of these two (and possibly more!) strategies in a single metric doesn't seem like a trivial job to me and IMO it needs a wider discussion in a separate issue instad of copying the .NET metric.

Some questions this topic immediately sparks:
- Do we know what are the exact semantics of `db.client.connections.wait_time`?
- Do other HTTP stacks have an equivalent metric? (I can't find anything in the linked netty and okhttp docs)
- What are the connection dispatching strategies of other stacks?
- Are we sure that in case of "coupled" implementations, where `wait_time == connection creation time` this is really a pool metric?
- Do we want to introduce connection spans (and possibly dns, tcp, tls subspans)? If yes how would they relate to this new metric in case of coupled and decoupled strategies?
- What happens if an implementation switches the strategy it's using? (Like it happened between .NET 5 and .NET 6)

Reply:
Thanks for the context!

I think  the key point you're raising which is also part of a bigger discussion is whether we want to have a set of generic metrics describing connections.

See #703, #454, #756 - this is no consensus in the community whether we should do it, but my preference is a generic set. See https://github.com/open-telemetry/semantic-conventions/issues/703#issuecomment-1958803934 for reasoning.

Option 1: generic metrics (imaginary)
- `connection.client.connect_duration` captures connection attempt duration and result
- `connection.pool.request_duration` captures time it takes to obtain connection from the connection pool

In case of 'coupled' mode, you'd capture the former but not the latter. In case of `decoupled` you'd capture both. 
.NET specific documentation for these metrics would describe .NET specifics. .NET can add extra attributes to those. 

Option 2: custom metrics for everything
- `dotnet.http.client.request.time_in_queue` captures whatever .NET wants to put there. But also  `dotnet.signalr.connection.connect_time` ? `microsoft.sql.client.connection.connect_time` ?
- `db.client.connections.wait_time` captures whatever application server puts into [JDBC connection pool JMX bean](https://www.ibm.com/docs/en/was/8.5.5?topic=pdo-jdbc-connection-pool-counters)
- when I implement connection metrics in Azure AMQP connections, I'd call them `azure.amqp.connection.duration`, etc.

Some of them would be precise like .NET one, others vague because they are cross-language, cross-client-lib definition anyway.

Having common metrics would allow users to easily switch between different technologies and understand the basics about the connections in they system independently of tech they use. I would be able to understand connection error rate, if they are pooled or not, how client libraries manages them to some extent.
They would lack specifics and sometimes could look too vague and abstract.

Having custom metrics for everything would result in different dashboard, alerts, queries, bigger mental load to support microservices, poor quality metrics due to lack of guidance, etc.
They might be overly-precise and expose users to internal implementation details they rarely care about.

I think the right solution is the middle-ground:
- some metrics are generalized. We need to decide which. I think socket connection level is quite reasonable since sockets are common technology shared by almost every language and library. These metrics are for basic and general analysis.
- some metrics are specific to the tech/system - they are for advanced users who need go deeper into the tech they are using.

These options are the same ones we're discussing here #760 and the litmus test we seem to lean towards to is if generic dashboards/alerts make sense. I'd say they do for socket connection metrics.

-------------
As you can see I don't have ready-to-use answers to your questions.

Getting back to the `http.client.request.time_in_queue`. It seems this metric future is uncertain
- Option 1: there will be a generic metric (metrics) conveying the same information
- Option 2: it will be renamed to `dotnet.http.client.request.time_in_queue` to emphasize it's .NET specifics (which can coexist with Option 1)
- Option 3: the metric will stay the same, but as part of .NET documentation, not yamlified and not shareable within otel.

I think the path forward would be to start with Option 3 and continue the discussion on Option 2 (or 1) separately within the .NET repo/community. I can create issues to follow up. LMK what  you think.

Reply:
> I think the path forward would be to start with Option 3 and continue the discussion on Option 2 (or 1)

I agree with this plan. Regarding option 1 VS option 2, I don't have an opinion since I lack knowledge about other stacks. All I wanted to highlight in the context of this PR is that this particular metric can't be adapted into the standard as-is because there are dragons.

Reply:
Alright, I went ahead and removed `http.client.request.time_in_queue` from the yaml. It is hardcoded in the .NET metrics markdown and won't be generated automatically anymore.